source-code/sdk-js/src/helpers/ast.test.ts

```diff
@@ -0,0 +1,215 @@
+            }
+        `
+		// @ivanohofer - what do you think of the type casting below?
+		const inputAst = parse(input, acornOptions) as unknown as Program
+		const result = wrapVariableDeclaration(inputAst, "blue", "wrapFn")
+		await expect(() => result).rejects.toThrowError("Couldn't find variable declarator.")
+	})
+	test("Doesn't manipulate inputs", async () => {
+		const input = `
+            export const load = () => {
+                return { data: true }
+            }
+        `
+		const inputAst = parse(input, acornOptions) as unknown as Program
+		const inputAstClone = structuredClone(inputAst)
+		await wrapVariableDeclaration(inputAst, "load", "wrapFn")
+		expect(inputAst).toEqual(inputAstClone)
+	})
+})
+
+describe("ast - insertAst", () => {
+	test("Inserts a simple import statement, before", async () => {
+		const insertion = `
+            import { wrapFn } from 'some/path'
+        `
+		const source = `
+            export const load = wrapFn(() => {
+                return { data: true }
+            })
+        `
+		const expected = `
+            import { wrapFn } from 'some/path'
+
+            export const load = wrapFn(() => {
+            return { data: true }
+            })
+        `
+		const insertionAst = parse(insertion, acornOptions) as unknown as Program
+		const sourceAst = parse(source, acornOptions) as unknown as Program
+		const expectedAst = parse(expected, acornOptions) as unknown as Program
+		const resultAst = await insertAst(sourceAst, insertionAst.body[0], { before: ["body", "0"] })
+		const expectedCode = generate(expectedAst)
+		const resultCode = generate(resultAst)
+		expect(resultCode).toEqual(expectedCode)
+	})
+	test("Inserts a load function, after", async () => {
+		const insertion = `
+            export const load = wrapFn(() => {
+                return { data: true }
+            })
+        `
+		const source = `
+            import { wrapFn } from 'some/path'
+        `
+		const expected = `
+            import { wrapFn } from 'some/path'
+
+            export const load = wrapFn(() => {
+            return { data: true }
+            })
+        `
+		const insertionAst = parse(insertion, acornOptions) as unknown as Program
+		const sourceAst = parse(source, acornOptions) as unknown as Program
+		const expectedAst = parse(expected, acornOptions) as unknown as Program
+		const resultAst = await insertAst(sourceAst, insertionAst.body[0], { after: ["body", "0"] })
+		const expectedCode = generate(expectedAst)
+		const resultCode = generate(resultAst)
+		expect(resultCode).toEqual(expectedCode)
+	})
+	test("Inserts code into wrap fn, before", async () => {
+		const source = `
+            import { wrapFn } from 'some/path'
+
+            export const load = wrapFn()
+        `
+		const expected = `
+            import { wrapFn } from 'some/path'
+
+            export const load = wrapFn(() => {
+                return { data: true }
+            })
+        `
+		const insertionAst = {
```

Comment:
why did you create the AST manually here?

Reply:
There was something strange happening in the process of parsing the string. The AST was not exactly representing what I was passing in as a string...

Reply:
This sounds like a bug or not?